File: src/payloads.ts
import {
  getDefaultProcessSharedBufferPrimitives,
  ProcessSharedBuffer,
} from "knitting/process-shared-buffer";

const BYTE_FILL_VALUES = [0xAB, 0xBC, 0xCD, 0xDE] as const;

export const makeBytePayloads = (bytes: number): Uint8Array[] =>
  BYTE_FILL_VALUES.map((fill) => new Uint8Array(bytes).fill(fill));

export const makePsbPayloads = (bytes: number): ProcessSharedBuffer[] => {
  const primitives = getDefaultProcessSharedBufferPrimitives();
  return BYTE_FILL_VALUES.map((fill) => {
    const psb = ProcessSharedBuffer.create(bytes, primitives);
    psb.bytes().fill(fill);
    return psb;
  });
};

export const powerOfTwoBytes = (minBytes: number, maxBytes: number): number[] => {
  const sizes: number[] = [];
  for (let bytes = minBytes; bytes <= maxBytes; bytes *= 2) sizes.push(bytes);
  return sizes;
};

File: src/payloads.rs
pub const BYTE_FILL_VALUES: &[u8] = &[0xAB, 0xBC, 0xCD, 0xDE];

pub fn make_byte_payloads(bytes: usize) -> Vec<Vec<u8>> {
    BYTE_FILL_VALUES.iter().map(|&fill| vec![fill; bytes]).collect()
}

pub fn power_of_two_sizes(min: usize, max: usize) -> Vec<usize> {
    let mut sizes = Vec::new();
    let mut n = min;
    while n <= max {
        sizes.push(n);
        n *= 2;
    }
    sizes
}
